### PR TITLE
Support Rails 7.0

### PR DESCRIPTION
**Pull request by ofedoren**

### CODE REVIEW

File: app/models/concerns/realm_theme.rb [hunk 1]
```diff
@@ -7,8 +7,9 @@ module RealmTheme
     Realm.class_eval do
       _validators.delete(:realm_type)
       _validate_callbacks.each do |callback|
-        if callback.raw_filter.respond_to? :attributes
-          callback.raw_filter.attributes.delete :realm_type
+        filter_method = callback.respond_to?(:raw_filter) ? :raw_filter : :filter
```
Comment (ekohl) [inline app/models/concerns/realm_theme.rb:10]: What's the difference here? Is this something for compatibility with old/new versions that we can drop at some point?

Comment (ofedoren) [reply app/models/concerns/realm_theme.rb:10]: Yeap, that's just for the compatibility. Although, we tend to require Foreman version in engines, I'd say we don't even need `callback.respond_to?(:raw_filter)` compatibility fix, just bump required version to >= 3.14 and use `filter` right away.

`raw_filter` was removed in Rails 7, `filter` seems to be the same:
 - https://github.com/rails/rails/blob/6-1-stable/activesupport/lib/active_support/callbacks.rb#L305
 - https://github.com/rails/rails/blob/7-0-stable/activesupport/lib/active_support/callbacks.rb#L299

Comment (evgeni) [reply app/models/concerns/realm_theme.rb:10]: 6.1 aliases raw_filter to filter, so we can use that already and don't have to do anything special here? Or am I reading the links wrong? 6.1 has been a requirement for a long time.

Comment (ekohl) [reply app/models/concerns/realm_theme.rb:10]: Reading the code, I see for 6.1:
```ruby
        def filter; @key; end
        def raw_filter; @filter; end
```

And 7.0:
```ruby
        attr_reader :chain_config, :filter
```

So 6.1 has `filter` read `@key` while 7.0 reads `@filter`. In other words, `filter` changes behavior between the versions.

Comment (evgeni) [reply app/models/concerns/realm_theme.rb:10]: 🤯
Did I mention how much I like Rails?

**Review by ekohl — APPROVED** [review]
I'm good with the compatibility approach, though I'd appreciate a code comment above it mentioning it's for Rails 6.1 & 7.0 compatibility so if we forget and later see it that we easily know.

@evgeni what do you prefer? hard cut or compatibility?

**Comment by evgeni** [comment]
> I'm good with the compatibility approach, though I'd appreciate a code comment above it mentioning it's for Rails 6.1 & 7.0 compatibility so if we forget and later see it that we easily know.
> 
> @evgeni what do you prefer? hard cut or compatibility?

The filter method thing is the only one, right? Then I'd go with compat as it doesn't seem super complex.

**Review by evgeni — APPROVED** [review]

**Pull request merged**